### PR TITLE
Gidk 1413 stage| Rohit | Discount and Taxes show on debtor creditor groups and accounts

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 1]
```diff
@@ -475,7 +475,12 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
     };
     /** No results found label for dynamic search */
     public noResultsFoundLabel = SearchResultText.NewSearch;
-
+    /** account's applied tax list */
+    public accountAssignedApplicableTaxes: string[] = [];
+    /** account's applied tax list */
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:480]: both variables are for accounts' applied tax list?

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html [hunk 2]
```diff
@@ -450,6 +450,42 @@ <h3>Address {{i+1}}</h3>
...
 
     </div>
+    <div class="form" *ngIf="isDebtorCreditor && isOtherSelectedTab">
+        <div class="mrB2 selectBox">
+            <div class="form-group ">
+                <label>Default Discount <span
+                    class="primary_clr"></span></label>
+                <div class="clearfix">
+                <span>
+                        <sh-select class="pull-left" [options]="discountList" [isFilterEnabled]="true" [forceClearReactive]="forceClearDiscount$ | async"  [multiple]="true" [(ngModel)]="selectedDiscounts" [placeholder]="'Select discounts'" style="width: calc(100% - 93px);"></sh-select>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html:460]: remove inline css

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 3]
```diff
@@ -217,6 +224,7 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
                         _.uniq(this.selectedDiscounts);
                     });
                 }
+                console.log('accountInheritedDiscounts', this.accountInheritedDiscounts, this.selectedDiscounts);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:227]: remove console

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 4]
```diff
@@ -1434,4 +1442,26 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
         }
         return false;
     }
+
+/**
+ * To apply discount in accounts
+ *
+ * @memberof AccountUpdateNewDetailsComponent
+ */
+public applyDiscounts(): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1451]: format code

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 5]
```diff
@@ -1434,4 +1442,26 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
+        if (this.accountDetails) {
+            this.activeAccountName = this.accountDetails.uniqueName;
+        } else {
+            this.activeAccount$.pipe(take(1)).subscribe(a => this.activeAccountName = a.uniqueName);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1455]: dont' use single character variable

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html [hunk 6]
```diff
@@ -145,14 +131,50 @@
...
 </div>
+
+<div class="horizontal-form mr1" *ngIf="isDebtorCreditorGroups">
+    <div class="mrB2 selectBox">
+        <div class="form-group ">
+            <label>Default Discounts <span
+                        class="primary_clr"></span></label>
+            <div class="clearfix">
+                <span>
+                        <sh-select class="pull-left" [options]="discountList" [isFilterEnabled]="true" [forceClearReactive]="forceClearDiscount$ | async"  [multiple]="true" [(ngModel)]="selectedDiscounts" [placeholder]="'Select discounts'" style="width: calc(100% - 93px);"></sh-select>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html:142]: remove inline css

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html [hunk 7]
```diff
@@ -450,6 +450,42 @@ <h3>Address {{i+1}}</h3>
         </div>
 
     </div>
+    <div class="form" *ngIf="isDebtorCreditor && isOtherSelectedTab">
+        <div class="mrB2 selectBox">
+            <div class="form-group ">
+                <label>Default Discount <span
+                    class="primary_clr"></span></label>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html:457]: Avoid snake_case in class names

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html (hunk 2 above)
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html:460]: Avoid inline styles

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 8]
```diff
@@ -31,8 +31,9 @@ import { IGroupsWithAccounts } from 'apps/web-giddh/src/app/models/interfaces/gr
 import { IFlattenGroupsAccountsDetail } from 'apps/web-giddh/src/app/models/interfaces/flattenGroupsAccountsDetail.interface';
 import { DbService } from 'apps/web-giddh/src/app/services/db.service';
 import { IDiscountList } from 'apps/web-giddh/src/app/models/api-models/SettingsDiscount';
-import { AssignDiscountRequestForAccount } from 'apps/web-giddh/src/app/models/api-models/ApplyDiscount';
+import { AssignDiscountRequestForAccount, ApplyDiscountRequestV2 } from 'apps/web-giddh/src/app/models/api-models/ApplyDiscount';
 import { SettingsDiscountActions } from 'apps/web-giddh/src/app/actions/settings/discount/settings.discount.action';
+import { element } from '@angular/core/src/render3';
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:36]: Avoid using nested imports
Please check if it is used

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts (hunk 3 above)
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:227]: Avoid consoles

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts (hunk 4 above)
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1451]: Do run formatter just on this method

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html [hunk 9]
```diff
@@ -145,14 +131,50 @@
     </ng-container>
     <!-- taxes container -->
 </div>
+
+<div class="horizontal-form mr1" *ngIf="isDebtorCreditorGroups">
+    <div class="mrB2 selectBox">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html:136]: Avoid use of camelCase class names

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html [hunk 10]
```diff
@@ -145,14 +131,50 @@
...
+        <div class="form-group ">
+            <label>Default Discounts <span
+                        class="primary_clr"></span></label>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html:139]: Avoid use of snake_case class names

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html (hunk 6 above)
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html:142]: Avoid use of [placeholder] for static label
Avoid inline styles

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts (hunk 8 above)
Comment (rohit-walkover) [reply apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:36]: yes it is used

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 11]
```diff
@@ -1443,16 +1442,16 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
- *
- * @memberof AccountUpdateNewDetailsComponent
- */
-public applyDiscounts(): void {
+    /**
+     * To apply discount in accounts
+     *
+     * @memberof AccountUpdateNewDetailsComponent
+     */
+    public applyDiscountselement(): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1450]: Please use proper camel capsing

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts (hunk 11 above)
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1450]: Avoid using incorrect camelCase in method names

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request closed without merging**